batching-core/src/main/java/com/flipkart/batching/core/BatchImpl.java

```diff
@@ -32,7 +32,10 @@
 
 public class BatchImpl<T extends Data> implements Batch<T> {
 
-    private DataCollection<T> dataCollection;
+    public DataCollection<T> dataCollection;
+
+    public BatchImpl() {
```

Comment:
please keep this constructor as package protected so that it is not misused otherwise. This will be used only by Stag